addons/im_livechat/models/discuss_channel_member.py

```diff
@@ -41,3 +41,8 @@ def _get_partner_data(self, fields=None):
                 } if self.partner_id.country_id else False
             return data
         return super()._get_partner_data(fields=fields)
+    
+    def _get_as_sudo_from_context_or_raise(self, channel_id):
+        if self.env.user.has_group("im_livechat.im_livechat_group_manager"):
+            return self._get_as_sudo_from_context(channel_id)
+        return super()._get_as_sudo_from_context_or_raise(channel_id) 
```

Comment:
If we keep it this way, it should be restricted to livechat too here.

But I don't really like this approach.
Calls to `_get_as_sudo_from_context_or_raise` on channel member should stay unchanged.
If we need a member, then it should return a member or raise.

But, places where it is just just to retrieve the channel should adapted (accessing channel_id on it and nothing else), to have a new method that gets or raise the channel, and this method would make more sense to have this override.

Also it should remove the need to override the controllers as is done currently.

Reply:
Note this is very similar to existing `_get_from_context_or_raise` on thread/channel.

Reply:
Temporarily added the channel type restriction here also.